shared/src/question-generators/propositional-logic/semanticEquivalence.ts

```diff
@@ -0,0 +1,207 @@
+    const { equivStatements, notEquivStatements } = createEquivStatements(random, expr, exprNames)
+    const { answers, correctAnswerIndex } = getAnswers(
+      random,
+      equivStatements,
+      notEquivStatements,
+      equivBool,
+    )
+    const inputExpressions = createInputExpressions(expr, exprNames)
+
+    const question: MultipleChoiceQuestion = {
+      type: "MultipleChoiceQuestion",
+      name: SemanticEquivalence.name(lang),
+      path: permalink,
+      text: t(translations, lang, "text", [
+        inputExpressions,
+        t(translations, lang, equivBool.toString()),
+      ]),
+      answers,
+      allowMultiple: true,
+      feedback: minimalMultipleChoiceFeedback({ correctAnswerIndex }),
+    }
+
+    return { question }
+  },
+}
+
+/**
+ * Creates five answers, combining true and not true statements.
+ * Computes the indices for the correct answers.
+ * @param random
+ * @param equivStatements - all possible true equivalences
+ * @param notEquivStatements - all possible false equivalences
+ * @param equivBool - If questions asks to determine true/false equivalences
+ */
+function getAnswers(
+  random: Random,
+  equivStatements: string[],
+  notEquivStatements: string[],
+  equivBool: boolean,
+) {
+  // If there are more equivStatements, include as many as necessary such that non-equivalences reach 5.
+  // Otherwise, include enough equivalences and fill with non-equivalences.
+  const answers = random.subset(
+    equivStatements,
+    min(equivStatements.length, random.int(max(5 - notEquivStatements.length, 1), 4)),
+  )
+  answers.push(...random.subset(notEquivStatements, 5 - answers.length))
+  random.shuffle(answers)
+
+  const correctAnswerIndex: number[] = answers
+    .map((x, i) => (equivStatements.includes(x) === equivBool ? i : -1))
+    .filter((index) => index !== -1)
+
+  return { answers, correctAnswerIndex }
+}
+
+/**
+ * Creates a tex block for each expression
+ * @param expr - list of boolean expressions
+ * @param exprNames - each expression gets a different name
+ */
+function createInputExpressions(expr: SyntaxTreeNodeType[], exprNames: string[]) {
+  let inputExpressions: string = ""
+  for (let i = 0; i < expr.length; i++) {
+    inputExpressions += `\\[ ${exprNames[i]}=${expr[i].toString(true)} \\] `
+  }
+  return inputExpressions
+}
+
+/**
+ * Creates n boolean expressions.
+ * @param random
+ * @param numLeaves - size of the tree +- random.int(-1, 2) to have different sizes
+ * @param varNames - possible variable names in the expression
+ * @param numVars - number of different variables
+ * @param numDiffExpr - number of expressions which aren't equivalent
+ * @param numExpr
+ */
+function getRandomExpressions(
+  random: Random,
+  numLeaves: number,
+  varNames: string[],
+  numVars: number,
+  numDiffExpr: number,
+  numExpr: number,
+) {
+  const expr: SyntaxTreeNodeType[] = [
+    generateRandomExpression(random, numLeaves + random.int(-1, 2), varNames.slice(0, numVars)),
```

Comment:
Can be simplified if `expressionsDifferent` is changed according to the comment.
Initialize to the empty list and start the for loop at `i=0`.